[ADD] l10n_fr_naf_ape from community-data-files

clonedagain opened the pull request:
Module moved from the OCA repository community-data-files.
First step to fix #1 .
The module works as-is on v8. README file added.

Commit:
[ADD] l10n_fr_naf_ape: add French official NAF partner categories and APE activity

Commit:
[REF] add a hint to a server branch containing a usefull feature in l10n_fr_ape_naf

Commit:
[REF] Cosmetics

Commit:
[MERGE] improvements from the 6.0 branch

Commit:
[IMP] Find NACE categories by identifier rather than by name

Commit:
[MERGE] improved l10n_fr_naf_ape

Commit:
[IMP] Update to v7.0 coding standard

Commit:
[IMP] PEP8

Commit:
[IMP] Place the NAF field next on the left side of the view

Commit:
[FIX] Force the web client to call name_get() on NAF categories instead of relying on search/read

clonedagain commented:
Updated to include the revision history of the 2 modules which I had neglected to include.

yvaucher commented:
:+1:

legalsylvain commented:
Hi,
Thanks for porting this module. 2 thinks :
- It's better to make 2 PR for two modules;
- For l10n_eu_nace, I don't see the reason to include an european module into fr localization;
  
  :question:  CC : @alexis-via

yvaucher commented:
@legalsylvain Do you suggest to add an european repository?

legalsylvain commented:
In fact, if it is a EU project, it will be usefull for german, italian teams.
So, I would like to avoid forking that module into all eu localization. 
Other solution is to let this module into community-data-files, it isn't ?
What do you think ?
CC : OCA: @sebastienbeau, @gurneyalex, @alexis-via  ...

clonedagain commented:
The French part extends the EU part so I'd rather have both in the same 
repo, otherwise the French part is not installable.
I have no idea what's best, opinions are welcome.

Le 18/12/2014 13:55, Sylvain LE GAL a écrit :

> In fact, if it is a EU project, it will be usefull for german, italian 
> teams.
> So, I would like to avoid forking that module into all eu localization.
> Other solution is to let this module into community-data-files, it isn't ?
> What do you think ?
> CC : OCA: @sebastienbeau https://github.com/sebastienbeau, 
> @gurneyalex https://github.com/gurneyalex, @alexis-via 
> https://github.com/alexis-via ...
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/OCA/l10n-france/pull/3#issuecomment-67481981.

gurneyalex commented on __unported__/l10n_eu_nace/__openerp__.py:
```diff
@@ -0,0 +1,54 @@
+{
+    "name": "European NACE partner categories",
+    "version": "2.0",
+    'author': u'Numérigraphe SARL, Sistheo',
```
Could you please add "Odoo Community Association (OCA)" to the list of authors?

gurneyalex commented on __unported__/l10n_eu_nace/__openerp__.py:
```diff
@@ -0,0 +1,54 @@
+    NACE_REV2_<language code>.csv
+Then update the LANGS constant in the script "make_data.py" and run it to
+refresh the OpenERP data files. Finally, upgrade the module to load the data.
+""",
```
I suggest moving the description to README.rst and using the OCA template.

yvaucher replied:
For now those modules are in `__unported__`

gurneyalex commented on __unported__/l10n_fr_naf_ape/__openerp__.py:
```diff
@@ -0,0 +1,41 @@
+{
+    'name': 'French NAF partner categories and APE code',
+    'version': '1.0',
+    'author': u'Numérigraphe SARL',
```
Could you add "Odoo Community Association (OCA)" to the list of authors?

gurneyalex commented:
maybe we could use https://github.com/OCA/community-data-files for the `l10n_eu_nace` addon?

pedrobaeza commented:
Isn't already there but unported (https://github.com/OCA/community-data-files/tree/8.0/__unported__/l10n_eu_nace)? I think that one should be "OCA level", not localization level.

clonedagain commented:
Thanks all, I'll update this proposal based on your suggestions.

`l10n_eu_nace` can live on in `community-data-files` if it fits more there. I'll add a warning (EDIT: in the description) so interested users can install it.

Commit:
Add 'l10n_fr_naf_ape/' from commit '09c25aca25720f58e6973d8d46e71b00dd3b54b6'

git-subtree-dir: l10n_fr_naf_ape
git-subtree-mainline: 840b5860e704185abc733867c597173b2b7dc001
git-subtree-split: 09c25aca25720f58e6973d8d46e71b00dd3b54b6

Commit:
[IMP] licence + README

Commit:
[FIX] Travis: install community-data-files

clonedagain commented:
Done. I've also tested the modules on v8: they work unchanged.

Travis fails here because the `community-data-files` part is `__unported__`.
Can someone please review https://github.com/OCA/community-data-files/pull/3 ?

gurneyalex commented on l10n_fr_naf_ape/README.rst:
```diff
@@ -0,0 +1,46 @@
+To install this module, you may either:
+
+* download it from the Odoo app store: https://www.odoo.com/apps?search=l10n_eu_nace
+* OR download and install the code of the OCA project *Community data files*: https://github.com/numerigraphe/community-data-files
```
could you use https://github.com/OCA/community-data-files ?

gurneyalex commented on l10n_fr_naf_ape/partner.py:
```diff
@@ -0,0 +1,34 @@
+    _columns = {
+        'ape_id': fields.many2one(
+            'res.partner.category', 'APE',
+            help="If the partner is a French company, enter it's official "
```
s/it's/its/

gurneyalex commented on l10n_fr_naf_ape/partner.py:
```diff
@@ -0,0 +1,34 @@
+            help="If the partner is a French company, enter it's official "
+                 "main activity in this field. The APE is chosen among the "
+                 "NAF nomenclature."),
+    }
```
question: I'm wondering if this ape_id should be added to the fields returned by res_partner._commercial_fields() to be copied to contacts (cf. https://github.com/odoo/odoo/blob/8.0/openerp/addons/base/res/res_partner.py#L446)

clonedagain replied:
At the moment I don't think so, since the field is hidden for non-company contacts. But if someone has a use for it I wouldn't block their PR.

gurneyalex commented:
overall :+1: once points above have been resolved

legalsylvain commented:
> Thanks all, I'll update this proposal based on your suggestions.
> l10n_eu_nace can live on in community-data-files if it fits more there. I'll add a warning (EDIT: in the description) so interested users can install it.

@clonedagain : Just my PoV, but I think the warning is unecessary. All dependencies are written in Travis File. So information is redundant. It should be usefull if it was a non OCA repository.

legalsylvain commented:
:+1: Code review. No test.
Thanks for porting this module and for the refactoring into to various repositories.

yvaucher commented:
needs fixing as some remarks from @gurneyalex still needs to be addressed

Commit:
[IMP] typo + translation (FR)

clonedagain commented:
Fixed I think, Travis should succeed once OCA/community-data-files#3 gets merged.

clonedagain commented:
Build passes if you force it: https://travis-ci.org/OCA/l10n-france/builds/54259385
Can we merge please ?

yvaucher commented:
Travis restarted

yvaucher commented:
Ok merged
